### PR TITLE
Update storage examples, add storage section to readme

### PR DESCRIPTION
**Pull request by mziccard**
This PR updates storage examples and adds one to the readme, as discussed in #164.
Feedback is appreciated!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -86,6 +87,45 @@ if (entity == null) {
 }
 ```
 
+Google Cloud Storage
+----------------------
+
+Google [Cloud Storage][cloud-storage] is a durable and highly available
```
Comment (jgeewax) [inline README.md]: Might want to be consistent and bracket the whole thing for GCS.

Comment (mziccard) [inline README.md]: Yes, looks better. I'll also update the Google Cloud Datastore link then.

File: README.md [hunk 2]
```diff
@@ -86,6 +87,45 @@ if (entity == null) {
...
+
+See the ``gcloud-java`` API [storage documentation][storage-api] to learn how to interact
+with the Cloud Storage using this Client Library.
+
+```java
+import com.google.gcloud.storage.Blob;
+import com.google.gcloud.storage.Storage;
+import com.google.gcloud.storage.StorageFactory;
+import com.google.gcloud.storage.StorageOptions;
+import java.nio.ByteBuffer;
```
Comment (aozarov) [inline README.md]: empty line between com.google to java.nio

File: README.md [hunk 3]
```diff
@@ -86,6 +87,45 @@ if (entity == null) {
...
+import java.nio.channels.WritableByteChannel;
+
+StorageOptions options = StorageOptions.builder().projectId(PROJECT_ID).build();
+Storage storage = StorageFactory.instance().get(options);
+byte[] content = readContent();
```
Comment (aozarov) [inline README.md]: Here and in package-info, I wonder if we need a better example that does not include 2 undefined methods "readContent" and "mergeContent"...

File: gcloud-java-examples/src/main/java/com/google/gcloud/examples/StorageExample.java [hunk 4]
```diff
@@ -142,12 +144,12 @@ public void run(Storage storage, BlobInfo... blobInfos) {
       if (blobInfos.length == 1) {
```
Comment (aozarov) [inline gcloud-java-examples/src/main/java/com/google/gcloud/examples/StorageExample.java]: Honestly, I think most of the changes in the file would look nicer if we change BlobAction and BlobsActionto have their own abstract methods and taking Blob or Bucket, Blob.... respectively.

Comment (mziccard) [inline gcloud-java-examples/src/main/java/com/google/gcloud/examples/StorageExample.java]: I agree actually. I just didn't want to modify too much the code. But I am happy to change!

File: README.md (hunk 3 above)
Comment (mziccard) [inline README.md]: I was wondering this as well. The only problem I see is that defining a byte array of content is not very meaningful for the user,  while using strings involve dealing with encoding, what do you think?

Comment (aozarov) [inline README.md]: If we both display content and also update/set content, then we probably need to encode/decode.
And as long as we do that consistently (e.g. using UTF_8) that would be a good example (I think).

File: gcloud-java-examples/src/main/java/com/google/gcloud/examples/StorageExample.java (hunk 4 above)
Comment (aozarov) [inline gcloud-java-examples/src/main/java/com/google/gcloud/examples/StorageExample.java]: Great, lets do it then.

**Comment by mziccard** [comment]
Any other comments on this?

**Comment by aozarov** [comment]
@mziccard Sorry didn't notice the new commits. looking at it now.

**Comment by aozarov** [comment]
Looks good. Thanks.

**Pull request merged**